astro-airflow-iris/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/settings.py

```diff
@@ -15,10 +15,13 @@
 # Class that manages storing KedroSession data.
 # from kedro.framework.session.store import BaseSessionStore
 # SESSION_STORE_CLASS = BaseSessionStore
-# Keyword arguments to pass to the `SESSION_STORE_CLASS` constructor.
-# SESSION_STORE_ARGS = {
-#     "path": "./sessions"
-# }
+
+# Setup for Experiment Tracking
+# The SQLite DB required for experiment tracking is stored by default
+# (supported from python >= 3.9 and Kedro-Viz 9.2.0) in the .viz folder
+# of your project. To store it in another directory, provide the keyword argument
+# `SESSION_STORE_ARGS` to pass to the `SESSION_STORE_CLASS` constructor.
+# SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2] / ".viz")}
```

Comment:
The description doesn't seem to match with the commented code. Is the `SESSION_STORE_CLASS` relevant?

Reply:
Hi @noklam , Could you please have a look at this - https://github.com/kedro-org/kedro-viz/issues/1951#issuecomment-2203426289

Also, do you suggest we do not specify anything about passing to the class ? like below -
# Setup for Experiment Tracking
# The SQLite DB required for experiment tracking is stored by default
# (supported from python >= 3.9 and Kedro-Viz 9.2.0) in the .viz folder
# of your project. To store it in another directory, provide the keyword argument
# `SESSION_STORE_ARGS`.
# SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2] / ".viz")}

Reply:
I can see the confusion. Why do you need to update all starters instead of just the `viz` related one? In general the expectation is uncommenting the settings would work, but in those case you have `BaseSessionStore` with  `SESSION_STORE_ARGS`.

This make sense for `viz` related:
```
# Class that manages storing KedroSession data.
# from kedro.framework.session.store import BaseSessionStore
# SESSION_STORE_CLASS = BaseSessionStore

# Setup for Experiment Tracking
# The SQLite DB required for experiment tracking is stored by default
# (supported from python >= 3.9 and Kedro-Viz 9.2.0) in the .viz folder
# of your project. To store it in another directory, provide the keyword argument
# `SESSION_STORE_ARGS` to pass to the `SESSION_STORE_CLASS` constructor.
# SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2] / ".viz")}
```

This confuse me for non-viz starter:
```python
from kedro_viz.integrations.kedro.sqlite_store import SQLiteStore  # noqa: E402

SESSION_STORE_CLASS = SQLiteStore

# Setup for Experiment Tracking
# The SQLite DB required for experiment tracking is stored by default
# (supported from python >= 3.9 and Kedro-Viz 9.2.0) in the .viz folder
# of your project. To store it in another directory, provide the keyword argument
# `SESSION_STORE_ARGS` to pass to the `SESSION_STORE_CLASS` constructor.
SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2])}

```

Reply:
The initial idea was to have comments in all the starters as someone could start from a non-viz starter but install viz separately and start using it. To avoid confusion, for non-viz starter, I will revert back the changes. 

For viz starter, I will have the following -

```
SESSION_STORE_CLASS = SQLiteStore

# Setup for Experiment Tracking
# The SQLite DB required for experiment tracking is stored by default
# (supported from python >= 3.9 and Kedro-Viz 9.2.0) in the .viz folder
# of your project. To store it in another directory, provide the
# keyword argument `SESSION_STORE_ARGS` to pass to the `SESSION_STORE_CLASS` constructor.
SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2])}

```

The reason I am not commenting - `SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2])}` is to support python 3.8 CI build [here](https://github.com/kedro-org/kedro-starters/actions/runs/10181808685/job/28173730832) as Kedro-Viz had a [bug](https://github.com/kedro-org/kedro-viz/pull/1915#issuecomment-2182777918) and now Viz does not support py3.8 . 

Thank you

Reply:
I think the manual update is still needed for existing users on >py3.8 if they intend to have session data stored in .viz location. 

For new users who are on >py3.8, they can refer to the comments mentioned and comment out the `SESSION_STORE_ARGS` if they want the default (.viz). 

In October, 2024 when py3.8 reaches end of life and starters stop supporting py3.8, we can comment out `SESSION_STORE_ARGS` so the new users will have the session data stored in .viz folder